Content.Server/Maps/GameMapManager.cs

```diff
@@ -217,20 +217,22 @@ private GameMapPrototype GetFirstInRotationQueue()
     {
         _log.Info($"map queue: {string.Join(", ", _previousMaps)}");
 
-        var eligible = CurrentlyEligibleMaps()
+        var eligibleList = CurrentlyEligibleMaps()
             .Select(x => (proto: x, weight: GetMapRotationQueuePriority(x.ID)))
-            .OrderByDescending(x => x.weight)
             .ToArray();
 
-        _log.Info($"eligible queue: {string.Join(", ", eligible.Select(x => (x.proto.ID, x.weight)))}");
+        var thresholdList = eligibleList
+            .Where(x => x.weight > 0)
+            .ToArray();
+
+        _log.Info($"eligible queue: {string.Join(", ", eligibleList.Select(x => (x.proto.ID, x.weight)))}");
 
         // YML "should" be configured with at least one fallback map
-        Debug.Assert(eligible.Length != 0, $"couldn't select a map with {nameof(GetFirstInRotationQueue)}()! No eligible maps and no fallback maps!");
+        Debug.Assert(eligibleList.Length != 0, $"couldn't select a map with {nameof(GetFirstInRotationQueue)}()! No eligible maps and no fallback maps!");
 
-        var weight = eligible[0].weight;
-        return eligible.Where(x => x.Item2 == weight)
-            .MinBy(x => x.proto.ID)
-            .proto;
+        // Choose a random map based on the threshold condition
+        var selectedList = thresholdList.Any() ? thresholdList : eligibleList;
+        return _random.Pick(selectedList).proto;
```

Comment:
Code doesn't really make sense to me:
1. Why are you getting a list with weights then immediately discarding it if the weight is <= 0
2. Why does it randomly pick from a *weighted* list?
3. Why use LINQ for Any instead of count.

Reply:
let me fix this rq :)

Reply:
Actually just noticed, while fixing this i accidentally forgot what the function was originally for HAHA, I'll close this and revisit later to completely recode, but i just want to let y'all know that function doesnt work properly